src/core/net/srp_client.cpp

```diff
@@ -68,13 +68,19 @@ void Client::HostInfo::Clear(void)
     SetState(kRemoved);
 }
 
-void Client::HostInfo::SetState(ItemState aState)
+bool Client::HostInfo::SetState(ItemState aState)
 {
-    if (aState != GetState())
-    {
-        LogInfo("HostInfo %s -> %s", ItemStateToString(GetState()), ItemStateToString(aState));
-        mState = MapEnum(aState);
-    }
+    bool didChange;
+
+    VerifyOrExit(aState != GetState(), didChange = false);
```

Comment:
thanks for handling this! 

should all state changes be considered to drop the response? For example: kRegistered -> kToRefresh could mean lease renewal and no change in services right?

Reply:
Note that only certain state changes are performed after msg is sent in 
```c++
anyChanged = ChangeHostAndServiceStates(kNewStateOnMessageTx, kForServicesAppendedInMessage);
```

 Check the `kNewStateOnMessageTx[]` array definition in `SendUpdate()`. The transition `kRegistered -> kToRefresh` cannot happen in this context and there wont matter in `anyChanged`.

```c++
    static const ItemState kNewStateOnMessageTx[]{
        /* (0) kToAdd      -> */ kAdding,
        /* (1) kAdding     -> */ kAdding,
        /* (2) kToRefresh  -> */ kRefreshing,
        /* (3) kRefreshing -> */ kRefreshing,
        /* (4) kToRemove   -> */ kRemoving,
        /* (5) kRemoving   -> */ kRemoving,
        /* (6) kRegistered -> */ kRegistered,
        /* (7) kRemoved    -> */ kRemoved,
    };
```

Reply:
Thanks for the clarification. Looks good